runtime/stdlib/account.go

```diff
@@ -670,21 +678,116 @@ func newAccountKeysGetFunction(
 					inter,
 					locationRange,
 					accountKey,
-					// public keys are assumed to be already validated.
-					func(
-						_ *interpreter.Interpreter,
-						_ interpreter.LocationRange,
-						_ *interpreter.CompositeValue,
-					) error {
-						return nil
-					},
+					assumePublicKeyIsValid,
 				),
 			)
 		},
 		sema.AccountKeysTypeGetFunctionType,
 	)
 }
 
+func newAccountKeysForEachFunction(
+	gauge common.MemoryGauge,
+	provider AccountKeyProvider,
+	addressValue interpreter.AddressValue,
+) *interpreter.HostFunctionValue {
+	address := addressValue.ToAddress()
+	// the AccountKey in `forEachKey(_ f: ((AccountKey): Bool)): Void`
+	callbackTypeParams := []sema.Type{sema.AccountKeyType}
+
+	return interpreter.NewHostFunctionValue(
+		gauge,
+		func(invocation interpreter.Invocation) interpreter.Value {
+			fnValue, ok := invocation.Arguments[0].(interpreter.FunctionValue)
+
+			if !ok {
+				panic(errors.NewUnreachableError())
+			}
+
+			inter := invocation.Interpreter
+			locationRange := invocation.LocationRange
+
+			newSubInvocation := func(key interpreter.Value) interpreter.Invocation {
+				return interpreter.NewInvocation(
+					inter,
+					nil,
+					[]interpreter.Value{key},
+					callbackTypeParams,
+					nil,
+					locationRange,
+				)
+			}
+
+			liftKeyToValue := func(key *AccountKey) interpreter.Value {
+				return NewAccountKeyValue(
+					inter,
+					locationRange,
+					key,
+					assumePublicKeyIsValid,
+				)
+			}
+
+			count := int(provider.AccountKeysCount(address))
+
+			var err error
+			var accountKey *AccountKey
+
+			for index := 0; index < count; index++ {
+				wrapPanic(func() {
+					accountKey, err = provider.GetAccountKey(address, index)
+				})
+				if err != nil {
+					panic(err)
+				}
+
```

Comment:
Just noticed this while reviewing the doc additions. 

Is there a particular reason why revoked keys are filtered out?
We might want to either remove the filtering or make it more obvious in the function name.

Not sure if it already does, but would be good for AccountKey to have an index.

Reply:
oh, I just made an assumption based on how I would expect key iteration to behave as a smart contract developer. probably should have checked in on that. @dsainati1 @SupunS any thoughts?

AccountKeys do have an index, btw

Reply:
I don't really know the answer here, it would be good to get some more input. I have added this as a topic to the next LDM. In the meantime we can maybe leave this PR open, there is no rush to get it in. Once a feature is shipped it is much harder to change

Reply:
hmm, given that `keys.get()` would also return revoked keys, I feel this should also include revoked keys. 
If not, like Bastian has said, we might want to rename the method to be more explicit about it.

I'm biased toward the first option because it supports both uses-cases (users can filter out if they want).

Reply:
ok, posted a poll on slack also and the overwhelming consensus is for revoked keys to be included. removed the filtering